lib/response.ex

```diff
@@ -1,5 +1,42 @@
 defmodule HTTPServer.Response do
-  defstruct status_code: 404, status_message: :not_found, resource: nil, headers: %{}, body: ""
-  @type status_code :: 200 | 404
-  @type status_message :: :ok | :not_found
+  defstruct status_code: nil, status_message: "", resource: nil, headers: %{}, body: ""
+
+  @type t :: %__MODULE__{
+          status_code: 200 | 404,
+          status_message: String.t(),
+          resource: String.t(),
+          headers: %{optional(String.t()) => any},
+          body: String.t()
+        }
```

Comment:
I made a similar formatting comment above. 
```suggestion
  @type t :: %__MODULE__{
    status_code: 200 | 404,
    status_message: String.t(),
    resource: String.t(),
    headers: %{optional(String.t()) => any},
    body: String.t()
  }
```